### DISCUSSION TITLE
Converting PNG with white background to JPG: results in a black background

### DISCUSSION THREAD

**Discussion by briankitt** [dlemstra/Magick.NET#710] [General] [answered]
Trying to convert a PNG with a white canvas to JPG.   It results in a black background.

I am on 7.17.0.1

Here is my code:
                    using (MagickImage reformat = new MagickImage(inputImage))
                    {
                        reformat.Format = MagickFormat.Jpeg;
                        reformat.BackgroundColor = MagickColors.White;
                        reformat.Settings.FillColor = MagickColors.White;
                        reformat.Settings.BackgroundColor = MagickColors.White;
                        reformat.FloodFill(MagickColors.White, 0, 0);
                        reformat.Write(ExportImage);
                    }
Below is my image.
![agent-8088RD](https://user-images.githubusercontent.com/3027960/89196410-c74cb280-d56f-11ea-972a-a4ad30f6a23f.png)

**Comment by dlemstra** [comment, marked as answer]
Your image has a transparent background. If you want to have a white background when you save the file as a jpeg you can color the alpha channel with the following code: 

```C#
using (MagickImage reformat = new MagickImage(inputImage))
{
    reformat.ColorAlpha(MagickColors.White);
    reformat.Write(ExportImage);
}
```

**Comment by briankitt** [comment]
Oh, that fixed it!   Thanks
